src/main/kotlin/domain/Cars.kt

```diff
@@ -0,0 +1,29 @@
+package domain
+
+class Cars(val cars: List<Car>) {
+
+    companion object CarFactory {
+        fun create(carNames: List<String>): Cars {
+            return Cars(carNames.map(::Car))
+        }
+    }
+
+    fun move(raceCount: Int, moveStrategy: MoveStrategy) {
+        repeat((0 until raceCount).count()) { round -> moveEachCars(round, moveStrategy) }
+    }
+
+    fun findWinners(): List<Car> {
+        val max = cars.maxOfOrNull(Car::position) ?: 0
```

Comment:
```suggestion
        val max = cars.maxOfOrNull(Car::position) ?: return emptyList()
```

maxOrNull은 컬렉션이 비어잇으면 null을 반환하는데 얼리 리턴을 해버려도 될거같네요

Reply:
오 최대값이 없으면 모든 자동차가 0으로 초기화 되니까, 모든 차가 우승이다 라고 할려고 했는데 생각해보니 초기값이 0 이라서 값이 없을 수 없겠네요. (널이 나온다는건 예외케이스겠네요)

오 저런식으로도 얼리리턴이 되는군요. 꿀팁 감삼다